OpenRA.Mods.Common/Traits/World/MusicPlaylist.cs

```diff
@@ -164,6 +164,12 @@ public void Play(MusicInfo music, Action onComplete)
 			Game.Sound.PlayMusicThen(music, onComplete);
 		}
 
+		public void SetBackgroundMusic(MusicInfo music)
+		{
+			currentBackgroundSong = music;
+			CurrentSongIsBackground = music != null;
```

Comment:
This line is related to the music playing, so shouldn't be set here.  Just to make this explicit (since the meaning of my earlier suggestions may have been lost in the confusion): SetBackgroundMusic should only have one line, which sets.  All the other state is managed by the (normal) music playback functions.